lib/far_mar/market.rb

```diff
@@ -0,0 +1,88 @@
+require 'csv'
+
+class FarMar::Market < FarMar::Base
+
+  MARKET_ARRAY = CSV.read("support/markets.csv")
+
+  attr_reader :name,
+              :address,
+              :city,
+              :county,
+              :state,
+              :zip
+
+    def initialize(attrs)
+      super(attrs[:id])           # @id = attrs[0].to_i
+      @name    = attrs[:name]     # @name = attrs[1]
+      @address = attrs[:address]  # @address = attrs[2]
+      @city    = attrs[:city]     # @city = attrs[3]
+      @county  = attrs[:county]   # @county = attrs[4]
+      @state   = attrs[:state]    # @state = attrs[5]
+      @zip     = attrs[:zip]      # @zip = attrs[6]
+    end
+
+  def self.all
+    # self.all - returns all rows of the CSV file as objects
+    new_market_array = []
+    # reads markets.csv file
+    # market_array = CSV.read("support/markets.csv")  # market_array = (self)read_file
+    MARKET_ARRAY.each do |market|  # can also do market_array.collect do |row|
```

Comment:
I see the comment about using `.collect`, but just wanted to point out that using `.collect` you wouldn't need to the `new_market_array` variable, so you could lose lines 26, 43, and 45.